apis/oauth/src/main/java/org/jclouds/oauth/v2/config/OAuthProperties.java

```diff
@@ -36,6 +36,22 @@
     */
    public static final String CREDENTIAL_TYPE = "jclouds.oauth.credential-type";
 
+   /**
+    * The client_credentials grant type allows for an optional "scope" to be included.
+    *
+    * @see <a href="https://tools.ietf.org/html/rfc6749#section-4.4.2">doc</a>
+    */
+   public static final String SCOPE = "jclouds.oauth.scope";
```

Comment:
Looks like this constant is used only in tests? Scopes in this API are configured per provider by binding an `OAuthScopes` instance. For example, in the [gogole-compute-engine provider](https://github.com/jclouds/jclouds/blob/master/providers/google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/config/GoogleComputeEngineHttpApiModule.java#L70-L73) we bind the scopes based on the permissions we want for the access token, according to [their api docs ](https://cloud.google.com/compute/docs/api/how-tos/authorization#user_auth_flow).

In the case of Azure, it looks like the parameter is not used (or that the common practice is to not send it) and the access token is granted a default scope. I've read somewhere that default scope in Azure, which is informed in the authentication response, is `user_impersonation`.

For consistency, I would suggest the following:

* Remove this property, as this should be configured by binding a `OAuthScopes` instance.
* Then either:
 * Mark the `OAuthScopes` as optional for injection, so a binding is not required. In this case, the Azure provider won't bind any scope and the authorization request won't send the "scopes" parameter.
 * Keep the `OAuthScopes` mandatory and, in Azure, configure the [AzureComputeHttpApiModule](https://github.com/ritazh/jclouds-labs/blob/azurecompute-arm/azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/config/AzureComputeHttpApiModule.java#L56) to bind the scopes as follows, in order to request a token with the Azure default scope grant (note that this approach will cause the authentication requests to be sent with the parameter `scope=user_impersonation`. I'm not very familiar with how OAuth works in Azure and I'm not sure this would be the right way to request the token):
  ```java
  bind(OAuthScopes.class).toInstance(OAuthScopes.SingleScope.create("user_impersonation"));
  ```